evalml/tests/component_tests/test_text_featurizer.py

```diff
@@ -460,3 +460,41 @@ def test_text_featurizer_sets_max_depth_1(mock_dfs):
     tf.fit(X)
     _, kwargs = mock_dfs.call_args
     assert kwargs["max_depth"] == 1
+
+
+@pytest.mark.parametrize("nones", [np.nan, pd.NA, None])
+def test_nan_allowed(nones):
```

Comment:
We should also add a test where there are nans in more than one column and in different indices